ksqldb-functional-tests/src/test/resources/historical_plans/filter_-_apply_filter_lambda_function_to_int_map/7.0.0_1614088997381/plan.json

```diff
@@ -0,0 +1,145 @@
+{
+  "plan" : [ {
+    "@type" : "ksqlPlanV1",
+    "statementText" : "CREATE STREAM TEST (ID STRING KEY, MAP MAP<STRING, INTEGER>) WITH (KAFKA_TOPIC='test_topic', KEY_FORMAT='KAFKA', VALUE_FORMAT='AVRO');",
+    "ddlCommand" : {
+      "@type" : "createStreamV1",
+      "sourceName" : "TEST",
+      "schema" : "`ID` STRING KEY, `MAP` MAP<STRING, INTEGER>",
+      "topicName" : "test_topic",
+      "formats" : {
+        "keyFormat" : {
+          "format" : "KAFKA"
+        },
+        "valueFormat" : {
+          "format" : "AVRO"
+        }
+      },
+      "orReplace" : false
+    }
+  }, {
+    "@type" : "ksqlPlanV1",
+    "statementText" : "CREATE STREAM OUTPUT AS SELECT\n  TEST.ID ID,\n  FILTER(TEST.MAP, (X, Y) => (((LEN(X) * 5) > 15) AND ((Y % 5) = 0))) LAMBDA\nFROM TEST TEST\nEMIT CHANGES",
```

Comment:
For my education: is the most-outer brackets required by the syntax? I.e. is 

```
(X, Y) => (LEN(X) * 5) > 15 AND (Y % 5) = 0 LAMBDA
```

Acceptable? If not, is

```
(X, Y) => ((LEN(X) * 5) > 15) AND ((Y % 5) = 0) LAMBDA
```

Acceptable?

Reply:
The outermost brackets are added by the test generator/parser (not sure which). The initial syntax for the test was `filter(map, (x,y) => (len(x) * 5 > 15) AND y % 5 = 0) AS lambda` and I just confirmed that it also works with `filter(map, (x,y) => len(x) * 5 > 15 AND y % 5 = 0) AS lambda` and `filter(map, (x,y) => ((len(x) * 5) > 15) AND y % 5 = 0) AS lambda`. Does that answer your question?

Reply:
Yup! That's good enough for me.